src/idpi/operators/time_operators.py

```diff
@@ -19,3 +19,38 @@ def time_rate(var: xr.DataArray, dtime: np.timedelta64):
     result = var.diff(dim="time") / (coord.diff(dim="time") / dtime)
     result.attrs = var.attrs
     return result
+
+
+def delta(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
+    """Compute difference for a given delta in time.
+
+    Parameters
+    ----------
+    field : xr.DataArray
+        Field that contains the input data.
+    dtime : np.timedelta64
+        Time delta for which to evaluate the difference.
+
+    Raises
+    ------
+    ValueError
+        if dtime is not multiple of the field time step.
+
+    Returns
+    -------
+    xr.DataArray
+        The Field difference for the given time delta.
+
+    """
+    coord = field.valid_time
+    condition = coord - coord[0] == dtime
+
+    try:
+        [index] = np.nonzero(condition.values)
+    except ValueError:
+        msg = "Provided dtime is not a multiple of the time step."
+        raise ValueError(msg)
+
+    result = field - field.shift(time=index.item())
+    result.attrs = field.attrs
```

Comment:
If we want to handle meta data strictly, we should perhaps introduce time bounds, and set the cell_methods attribute with respect to time (or code GRIB2-like attributes like typeOfStatisticalProcessing and lengthOfTimeRange).

Reply:
This sounds like a good idea but I don't know how this should be specified. In theory, we are not bound by grib2 semantics but it would be smart not to drift too far away.

Reply:
We should keep this issue in mind for further dicussion of the meta data model.

Reply:
I saw that you made a corresponding comment on the Confluence page dedicated to the meta data model.